apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/ErrorReporting/ErrorReportingExceptionLogger.cs

```diff
@@ -145,8 +145,8 @@ private HttpRequestContext CreateHttpRequestContext(HttpContext context)
             return new HttpRequestContext()
             {
                 Method = request?.Method?.ToString() ?? "",
-                Url = UriHelper.GetDisplayUrl(request) ?? "",
-                UserAgent = request?.Headers["User-Agent"] ?? "",
+                Url = request?.GetDisplayUrl() ?? "",
+                UserAgent = request?.Headers["User-Agent"].ToString() ?? "",
```

Comment:
The set for  UserAgent will throw if we try to assign it a null value so we either need to not set it or use the empty string.